src/lib.rs

```diff
@@ -72,10 +74,27 @@ pub use crate::{
 /// # }
 /// ```
 pub struct LockFile {
+    pub truncate_on_close: bool,
     locked: bool,
     desc: sys::FileDesc,
 }
 
+// Private functions
+impl LockFile {
+    fn new() -> Self {
+        #[allow(unused_unsafe)]
+        // not safe to get state of default truncation setting on no_std
+        Self {
+            desc: sys::uninitialized_fd(),
```

Comment:
I don't think having this "uninitialized" FD is worth it, it seems error prone. Having a state in which data is "uninitialized" should be avoided (even if not uninitialized in UB-sense). But why exactly does this method (new) exist?

Reply:
`LockFile::new` is a convenience function for calling here—https://github.com/brunoczim/fslock/pull/8/files/81c197d4dc1350561dee953167df57de06fd67fa#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L124-R143

Without having to `impl Default for FileDesc`